net.go

```diff
@@ -203,6 +203,47 @@ func (n *Net) Walk(r *net.IPNet, handle func(*net.IPNet, interface{}) bool) {
 	})
 }
 
+func walkMatch(p *node, key []byte, handle func(*net.IPNet, interface{}) bool) bool {
+	if p.internal != nil {
+		if !walkMatch(&p.internal.child[0], key, handle) {
+			return false
+		}
+
+		if key[p.internal.offset]&p.internal.bit > 0 {
```

Comment:
```diff
-		if key[p.internal.offset]&p.internal.bit > 0 {
+		if p.internal.offset >= len(key)-1 || key[p.internal.offset]&p.internal.bit > 0 {
```

Reply:
Oops, thanks! Fixed.
Can you want me to rebase this branch so that you can merge this without conflicts?